File: integrations/aws-strands/error-code-corpus/typescript/message_conditional.ts
/** A message that takes one of two literal branches. */
import { EventType, type BaseEvent } from "@ag-ui/core";

export function emit(detail: boolean): BaseEvent {
  return {
    type: EventType.RUN_ERROR,
    message: detail
      ? "The conditional true branch."
      : "The conditional false branch.",
    code: "MESSAGE_CONDITIONAL",
  };
}

File: integrations/aws-strands/error-code-corpus/python/message_conditional.py
"""A message that takes one of two literal branches."""

from ag_ui.core import EventType, RunErrorEvent


def emit(detail: bool) -> RunErrorEvent:
    return RunErrorEvent(
        type=EventType.RUN_ERROR,
        message=(
            "The conditional true branch."
            if detail
            else "The conditional false branch."
        ),
        code="MESSAGE_CONDITIONAL",
    )
